planner/core/find_best_task.go

```diff
@@ -812,10 +812,12 @@ func (ds *DataSource) findBestTask(prop *property.PhysicalProperty, planCounter
 				p: dual,
 			}, cntPlan, nil
 		}
-		canConvertPointGet := len(path.Ranges) > 0 && path.StoreType == kv.TiKV && ds.isPointGetConvertableSchema() &&
-			// to avoid the over-optimized risk, do not generate PointGet for plan cache, for example,
-			// `pk>=$a and pk<=$b` can be optimized to a PointGet when `$a==$b`, but it can cause wrong results when `$a!=$b`.
-			!ds.ctx.GetSessionVars().StmtCtx.UseCache
+		canConvertPointGet := len(path.Ranges) > 0 && path.StoreType == kv.TiKV && ds.isPointGetConvertableSchema()
+
+		if ds.ctx.GetSessionVars().StmtCtx.UseCache && canConvertPointGet {
```

Comment:
Maybe we can use `expression.MaybeOverOptimized4PlanCache(ctx, path.AccessConds)` rather than `ds.ctx.GetSessionVars().StmtCtx.UseCache`?

Reply:
updated